pkg/pipeline/templates/terraformed.go.tmpl

```diff
@@ -38,7 +38,7 @@ func (tr *{{ .CRD.Kind }}) GetParameters() (map[string]interface{}, error) {
 	}
 	base := map[string]interface{}{}
 	{{- if .CRD.ConfigureWithFn }}
-	{{ .CRD.ConfigureWithFn }}.Configure(base, meta.GetExternalName(tr))
+	{{ .CRD.ConfigureWithFn }}(base, meta.GetExternalName(tr))
```

Comment:
```suggestion
	// ConfigStoreBuilder is the one defined in CRD package.
	if cfg, ok := ConfigStoreBuilder[tr.GetTerraformResourceType()]; ok {
		// call the function in external name configuration
	}
```

I think the point of having a map was to be able to remove the function path as input. Would we be able to check the map and call the function?

Reply:
The problem with that is we would need to create an instance of ConfigStore and pass it to each controller at provider runtime as well. Currently, that map is initialized and used only during generation. 

It should not be a big deal and we can do it but I gave up doing that when I realize we would anyway need to pass string representation of reference types and extractor function, so, (we also synced on this) it is better to be consistent with external name function as well and take string representation of it as input which would mean no dependency on this ConfigStore object at runtime.

Reply:
With the latest refactoring, ignore the creating ConfigStore instance part. The only concern is consistency.

Reply:
Sounds good. It seems like the each configuration would work best with different format but that could be a pain to work with developers.